tools/config/__init__.py

```diff
@@ -792,26 +794,40 @@ def _generate_bootloader_build(self, rom_memories):
             if part.minaddr() != rom_start:
                 raise ConfigException("bootloader executable does not "
                                       "start at 0x%x" % rom_start)
+            regions = part.segments()
 
             # find the last valid address that's within rom_end and use that
             # to compute the bootloader size
-            end_address = None
-            for start, stop in part.segments():
-                if (stop < rom_end):
-                    end_address = stop
-                else:
-                    break
-            if end_address == None:
-                raise ConfigException("bootloader segments don't fit within rom region")
-            part_size = Config._align_ceiling(end_address, self.sectors) - rom_start
 
-            yield Region("bootloader", rom_start, part_size, False,
+            # we have multiple parts in bootloader. Treat each of them as
+            # a different region (BLP1, BLP2 ...)
+            if len(part.segments()) > 1:
+                end_address = None
+                part_count = 0
+                for start, stop in part.segments():
+                    part_count += 1
+                    if (stop < rom_end):
+                        end_address = stop
+                    else:
+                        break
+                    if end_address == None:
+                        raise ConfigException("bootloader segments don't fit within rom region")
```

Comment:
Please remove "region" from this exception, as ROM is not a region:
```suggestion
                        raise ConfigException("bootloader segments don't fit within rom")
```